self-core-impl/src/main/java/com/selfxdsd/core/tasks/ContractTasks.java

```diff
@@ -80,12 +81,9 @@ public Tasks ofContract(final Contract.Id id) {
             tasks = this;
         }else{
             //@checkstyle BooleanExpressionComplexity (10 lines)
-            final List<Task> ofContract = this.tasks.stream()
-                .filter(t -> t.assignee() != null && t.assignee()
-                    .username().equals(id.getContributorUsername())
-                    && t.project().repoFullName().equals(id.getRepoFullName())
-                    && t.project().provider().equals(id.getProvider())
-                    && t.role().equals(id.getRole()))
+            final List<Task> ofContract = StreamSupport
+                .stream(this.storage.tasks()
+                    .ofContract(id).spliterator(), false)
                 .collect(Collectors.toList());
```

Comment:
Actually this will throw stackoverflow exception with real implementation of storage...

Reply:
@criske But you don't need all this logic here. Just throw ISE, like in the other examples and say: "These are the Tasks of Contract X, you cannot see other Contract's tasks here"

Reply:
You're right :D